File: src/Problem016PowerDigitSum.java
void main() {
  String bigNumber = BigInteger.valueOf(2).pow(1000).toString();
  int digitsSum = 0;

  for (char digit : bigNumber.toCharArray()) {
    digitsSum += digit - '0';
  }

  IO.println("Digits Sum: " + digitsSum);
}

File: src/Problem016PowerDigitSum.kt
import java.math.BigInteger

fun main() {
  val bigNumber = BigInteger.valueOf(2).pow(1000)
  val digitsSum = bigNumber.toString()
    .map { it.toString().toInt() }
    .reduce { acc, i -> acc + i }

  println("Digits Sum: $digitsSum")
}
